src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java

```diff
@@ -231,16 +234,25 @@ public Filter convertFilter(FeatureTypeInfo info, Filter filter) {
         if (bb instanceof BBOX) {
             BoundingBox bounds = ((BBOX) bb).getBounds();
             LongLatConfiguration configuration = getLongLatConfiguration(info);
-            PropertyIsBetween longitudeFilter =
-                    createBetweenFilter(
-                            ff,
-                            configuration.longAttributeName,
-                            bounds.getMinX(),
-                            bounds.getMaxX());
-            PropertyIsBetween latitudeFilter =
-                    createBetweenFilter(
-                            ff, configuration.latAttributeName, bounds.getMinY(), bounds.getMaxY());
-            return ff.and(longitudeFilter, latitudeFilter);
+
+            return ff.and(getLongLatFilters(bounds, ff, configuration));
+        } else if (bb instanceof Or) {
+            // https://github.com/geosolutions-it/C105-2018-EMSA-VDS/issues/73
+            Or orFilter = (Or) bb;
+            // has multiple bboxes
+            List<Filter> propertyIsBetweenLists = new ArrayList<Filter>();
+            LongLatConfiguration configuration = getLongLatConfiguration(info);
+            for (Filter f : orFilter.getChildren()) {
+                // skip if this filter is not a BBOX
+                if (!(f instanceof BBOX)) continue;
```

Comment:
So if you pass a filter having a different operator it will be excluded?

Reply:
yes because in this function we are only looking to convert BBOX filters to simple between filter